src/branchandbound.jl

```diff
@@ -0,0 +1,81 @@
+using TaylorModels
+# univariate
+function enclose_BranchandBound(f::Function, dom::Interval{T}; order::Int = 10,
+                                tol::Number = 0.6) where{T}
+    x0 = Interval(mid(dom))
+    x = TaylorModel1(order, x0, dom)
+    return branchandbound(f(x-x0).pol, dom, tol)
+end
+# multivariate
+function enclose_BranchandBound(f::Function, dom::IntervalBox{T,N}; order::Int = 10,
+                                tol::Number = 0.6) where {T,N}
+    n = length(dom)
+    x0 = mid(dom)
+    set_variables(Float64, "x", order=2order, numvars=n)
+    x = [TaylorModelN(i, order, IntervalBox(x0), dom) for i=1:n]
+    return branchandbound(f(x...).pol, dom - x0, tol)
+end
+
+function branchandbound(p::Taylor1{T}, dom::Interval{W}, ϵ::Number) where{T,W}
+    return _branchandbound(p, dom, ϵ)
+end
+
+function branchandbound(p::TaylorN{T}, dom::IntervalBox{N,W}, ϵ::Number) where{T,N,W}
+    return _branchandbound(p, dom, ϵ)
+end
+
+function _branchandbound(p::Union{TaylorN{T},Taylor1{T}},
+                       dom::Union{IntervalBox{N,W},Interval{W}}, ϵ::Number) where {N,T,W}
+    K = 1
+    Rperv = evaluate(p, dom)
+    D1, D2 = bisect(dom)
+    D = [D1, D2]
+    R = [evaluate(p, D[i]) for i = 1:length(D)]
+    Rnext = _Rnext(R)
+    while  (Rperv.hi - Rnext.hi) <= ϵ*(Rnext.hi - Rnext.lo) &&
+           (Rperv.lo - Rnext.lo) <= ϵ*(Rnext.hi - Rnext.lo) && (K <= 1000)
```

Comment:
there is something odd here -- the same condition seems to be checked twice.

Reply:
Ok:)